index.js

```diff
@@ -75,13 +86,47 @@ function extend(target, source) {
   return target;
 }
 
+function prepareIndent(options) {
+  if (typeof options.contents !== 'string') {
+    throw new Error('"contents" option must be a string.');
+  }
+
+  var indentMatch;
+  if (options.indent === null) {
+    // auto detect by code
+    indentMatch = options.contents.match(/^([ \t])/);
+    options.indent =
+      (indentMatch) ? indentMatch[1] : options.defaultIndentValue;
+  } else if (typeof options.indent !== 'string') {
+    throw new Error('Unknown type of "indent" option value.');
+  }
+
+  var getContentsWithIndent;
+  (function (contents, indent) {
+    getContentsWithIndent = function (count) {
+      var indentStr = '';
+      for (var i=0; i<count; i++) {
+        indentStr += indent;
+      }
+      return contents
+        .replace(/\r\n/g, '\n')
+        .replace(/\r/g, '\n')
+        .replace(/\n/g, '\n' + indentStr)
+        .replace(/^[ \t]+$/gm, '');
+    };
+  })(options.contents.toString(), options.indent);
+
+  options.getContentsWithIndent = getContentsWithIndent;
+}
+
```

Comment:
Do you know of a module that does this? I am thinking that maybe [redent](https://npmjs.com/package/redent) would do the trick. I'll let you dig into the details if you are still interested in getting this merged.

Reply:
Nah, don't care about js world anymore, feel free do to with this anything you want.